src/main/java/me/xmrvizzy/skyblocker/skyblock/item/PriceInfoTooltip.java

```diff
@@ -156,54 +183,108 @@ public static String getInternalNameFromNBT(ItemStack stack) {
-                    CompletableFuture.runAsync(PriceInfoTooltip::downloadAvgPrices);
-                if (SkyblockerConfig.get().general.itemTooltip.enableLowestBIN || PriceInfoTooltip.firstRun)
+        skyblocker.scheduler.scheduleCyclic(() -> {
+            {
+                if ((SkyblockerConfig.get().general.itemTooltip.enableAvgBIN || minute == 0)
+                        && (oneDayAvgPricesJson == null || threeDayAvgPricesJson == null || minute % 5 == 0)) {
+                    SkyblockerConfig.Average avg;
+                    if (oneDayAvgPricesJson == null || threeDayAvgPricesJson == null || minute == 0) {
+                        avg = SkyblockerConfig.Average.BOTH;
+                    } else {
+                        avg = SkyblockerConfig.get().general.itemTooltip.avg;
+                    }
+                    switch (avg) {
+                        case ONE_DAY -> CompletableFuture.runAsync(PriceInfoTooltip::download1DayAvgPrices);
+                        case THREE_DAY -> CompletableFuture.runAsync(PriceInfoTooltip::download3DayAvgPrices);
+                        case BOTH -> {
+                            CompletableFuture.runAsync(PriceInfoTooltip::download1DayAvgPrices);
+                            CompletableFuture.runAsync(PriceInfoTooltip::download3DayAvgPrices);
+                        }
+                    }
+                }
+                if (SkyblockerConfig.get().general.itemTooltip.enableLowestBIN || minute == 0) {
                     CompletableFuture.runAsync(PriceInfoTooltip::downloadLowestPrices);
-                if (SkyblockerConfig.get().general.itemTooltip.enableBazaarPrice || PriceInfoTooltip.firstRun)
+                }
+                if ((SkyblockerConfig.get().general.itemTooltip.enableBazaarPrice || minute == 0) && (bazaarPricesJson == null || minute % 3 == 0)) {
                     CompletableFuture.runAsync(PriceInfoTooltip::downloadBazaarPrices);
-                if (SkyblockerConfig.get().general.itemTooltip.enableNPCPrice || PriceInfoTooltip.firstRun)
+                }
+                if ((SkyblockerConfig.get().general.itemTooltip.enableNPCPrice || minute == 0) && npcPricesJson == null) {
                     CompletableFuture.runAsync(PriceInfoTooltip::downloadNPCPrices);
-                if (SkyblockerConfig.get().general.itemTooltip.enableMuseumDate || PriceInfoTooltip.firstRun)
+                }
+                if ((SkyblockerConfig.get().general.itemTooltip.enableMuseumDate || minute == 0) && isMuseumJson == null) {
                     CompletableFuture.runAsync(PriceInfoTooltip::downloadIsMuseum);
+                }
+                minute++;
             }
-        };
-
-        firstRun = false;
-        Timer timer = new Timer("PriceInfoDownloader");
-        timer.scheduleAtFixedRate(repeatedTask, 0L, 1000L * 60L);
+        }, 1200);
     }
 
-    private static void downloadAvgPrices() {
+    private static void download1DayAvgPrices() {
```

Comment:
Both `download1DayAvgPrices` and `download3DayAvgPrices` are pretty much identical. Moving this common code into a separate function would increase readability and make the code easier to maintain.